tests/test_go_director.py

```diff
@@ -0,0 +1,598 @@
+                "key_type": "ip",
+                "evaluation_type": "score_confidence",
+            },
+            True,
+        ),
+        # Test case 2: Missing key
+        ({"key_type": "ip", "evaluation_type": "score_confidence"}, False),
+        # Test case 3: Invalid key type
+        (
+            {
+                "key": "192.168.1.1",
+                "key_type": "invalid_type",
+                "evaluation_type": "score_confidence",
+            },
+            False,
+        ),
+        # Test case 4: Invalid IP address
+        (
+            {
+                "key": "invalid_ip",
+                "key_type": "ip",
+                "evaluation_type": "score_confidence",
+            },
+            False,
+        ),
+        # Test case 5: Invalid evaluation type
+        (
+            {
+                "key": "192.168.1.1",
+                "key_type": "ip",
+                "evaluation_type": "invalid_evaluation",
+            },
+            False,
+        ),
+    ],
+)
+def test_validate_message_request(data, expected_result, mock_db):
+    go_director = ModuleFactory().create_go_director_obj(mock_db)
+    result = go_director.validate_message_request(data)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "ip, reporter, score, confidence, timestamp, "
+    "profileid_of_attacker, "
+    "expected_description, expected_threat_level",
+    [
+        # Test case 1: Basic test with valid data
+        (
+            "192.168.1.1",
+            "test_reporter",
+            0.5,
+            0.8,
+            1649445643,
+            "profile_192.168.1.1",
+            "attacking another peer:  " "(test_reporter). confidence: 0.8 ",
+            "medium",
+        ),
+        # Test case 2: Test with a different score and confidence
+        (
+            "10.0.0.1",
+            "another_reporter",
+            0.9,
+            0.6,
+            1649445644,
+            "profile_10.0.0.1",
+            "attacking another peer:  " "(another_reporter). confidence: 0.6 ",
+            "critical",
+        ),
+    ],
+)
+def test_set_evidence_p2p_report(
+    mock_db,
+    ip,
+    reporter,
+    score,
+    confidence,
+    timestamp,
+    profileid_of_attacker,
+    expected_description,
+    expected_threat_level,
+):
+    go_director = ModuleFactory().create_go_director_obj(mock_db)
+    go_director.trustdb.get_ip_of_peer.return_value = (timestamp, "")
+
+    go_director.set_evidence_p2p_report(
+        ip, reporter, score, confidence, timestamp, profileid_of_attacker
+    )
+
+    mock_db.set_evidence.assert_called_once()
+    call_args = mock_db.set_evidence.call_args[0][0]
+    assert call_args.attacker.value == ip
+    assert expected_description in call_args.description
+    assert call_args.threat_level == expected_threat_level
+
+
+def test_read_configuration(mock_db):
+    go_director = ModuleFactory().create_go_director_obj(mock_db)
+    go_director.read_configuration()
```

Comment:
here we should moch the returns value of this function, because if the default width in the config changed in the future, this function will fail